### PR TITLE
Handle http errors

### PR DESCRIPTION
**Pull request by MatthewBrand**
Changed all json.loads to use new method. Also taking into account
response.ok on all calls. Bamboo needed a bit of a different approach

### CODE REVIEW

File: User.py [hunk 1]
```diff
@@ -48,14 +48,16 @@ def storeSession(self, run_name):
...
         params = {"token": USER_TOKEN_STRING, "user": self.id}
-        response = requests.get("https://slack.com/api/users.info",
-                params=params)
-        user_obj = json.loads(response.text, encoding='utf-8')["user"]
+        response = requests.get("https://slack.com/api/users.info", params=params)
 
-        username = user_obj["name"]
-        real_name = user_obj["profile"]["real_name"]
+        parsed_message, isMessageOkay = parseSlackJSON(response)
+        if isMessageOkay:
```
Comment (mikeisman) [inline User.py:55]: We should log in the else case here with the response status

File: User.py [hunk 2]
```diff
@@ -95,3 +95,14 @@ def addExercise(self, exercise, reps):
...
+
+def parseSlackJSON(response):
+    isMessageOkay = False
+    try:
+        if response.ok:
+            parsed_message = json.loads(response.text, encoding='utf-8')
+    except:
+        print "Caught exception parsing response status: " + str(response.status_code) + ", text: " + response.text
+
+    isMessageOkay = response.ok & parsed_message["ok"]
```
Comment (mikeisman) [inline User.py]: this should be `&&`

File: User.py [hunk 3]
```diff
@@ -95,3 +95,14 @@ def addExercise(self, exercise, reps):
...
         return exercise["id"] in self.exercise_counts
 
+
+def parseSlackJSON(response):
+    isMessageOkay = False
+    try:
+        if response.ok:
+            parsed_message = json.loads(response.text, encoding='utf-8')
+    except:
+        print "Caught exception parsing response status: " + str(response.status_code) + ", text: " + response.text
```
Comment (mikeisman) [inline User.py]: Here you should `return null, False` so you don't get to the other part in the case that `response.ok` is True but we were unable to parse the json

File: slackbotExercise.py [hunk 4]
```diff
@@ -169,17 +169,24 @@ def selectUser(bot, exercise, all_employees, previous_winners):
...
+    if bot.debug:
+        print "fetchAllEmployees response: " + response.text
+
     all_employees = []
     if response.ok:
         if bot.debug:
             print "called bamboo successfully"
-        all_employees = json.loads(response.text)["employees"]
+
+        parsed_message = json.loads(response.text, encoding='utf-8')
```
Comment (mikeisman) [inline slackbotExercise.py]: This needs to be in the `try` in case it's not valid json

File: slackbotExercise.py [hunk 5]
```diff
@@ -463,6 +479,16 @@ def resetChallenges(bot):
 
     print "Reset users' challenges"
 
+def parseSlackJSON(response):
```
Comment (mikeisman) [inline slackbotExercise.py]: Don't repeat this twice, I think we should make an `HTTP.py` class and import it in the main script and in `User.py`

**Comment by mikeisman** [comment]
Overall, we need to figure out what happens if `isMessageOk` is false in each API call

File: User.py (hunk 2 above)
Comment (MatthewBrand) [inline User.py]: I don't think Python has an && operator. Doing & is a bitwise operator. So, evaluating that on true/false should be fine.

File: slackbotExercise.py (hunk 5 above)
Comment (MatthewBrand) [inline slackbotExercise.py]: Yeah, you're definitely right. I was feeling lazy :-1:

File: User.py (hunk 2 above)
Comment (mikeisman) [inline User.py]: Then use `and`. I _think_ it'll work fine with the bitwise operator, but we're looking to return strictly `True` or `False`

File: User.py (hunk 1 above)
Comment (MatthewBrand) [inline User.py:55]: I've added a print to the parse method to print out the response status if it's not okay.,

**Pull request merged**